### PR TITLE
Extend events channel to Runes

### PR DESCRIPTION
**Pull request by felipelincoln**
Extending the work done in PR #3137  to enable rune events.

### CODE REVIEW

File: src/index/event.rs [hunk 1]
```diff
@@ -17,4 +19,28 @@ pub enum Event {
...
+    rune_id: RuneId,
+    amount: u128,
+    outpoint: OutPoint,
+  },
+  RuneBurned {
+    block_height: u32,
+    txid: Txid,
+    rune_id: RuneId,
+    amount: u128,
+  },
```
Comment (felipelincoln) [inline src/index/event.rs:45]: The events fields are TBD

File: src/index/event.rs [hunk 2]
```diff
@@ -17,4 +19,28 @@ pub enum Event {
     old_location: SatPoint,
     sequence_number: u32,
   },
+  RuneEtched {
+    block_height: u32,
+    txid: Txid,
+    rune_id: RuneId,
+  },
+  RuneClaimed {
```
Comment (raphjaph) [inline src/index/event.rs:27]:
```suggestion
  RuneMinted {
```

**Review by raphjaph — CHANGES_REQUESTED** [review]
Good PR!

Sorry for getting to this so late. Would you mind merging in the latest changes and fixing the conflicts? Then I can have a final look

**Comment by kevinfaveri** [comment]
gm. hey @raphjaph in case @felipelincoln can we give a prio to adjusting this PR and merging before halving?

Almost everyone with custom infra relies on having those events or forking and adding stuff to Ord API....
Also makes me think, should this be scoped when you guys change Ord? And another question: maybe we think about a headless ord/ord version? That you can plug any DB rather than redb?

**Comment by raphjaph** [comment]
Ok, I'll try to fix this up

**Comment by raphjaph** [comment]
Fixed it up. @casey, @felipelincoln, @kevinfaveri this is ready for review

**Review by casey — APPROVED** [review]
LGTM

**Review by raphjaph — APPROVED** [review]

**Pull request merged**